dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/BasicTools.java

```diff
@@ -185,4 +189,15 @@ public String missingTitle(@ToolArg(name = "input", description = "input string"
         return input;
     }
 
+    @Tool(name = "cancellationTool", title = "Cancellable tool", description = "A tool that waits to be cancelled")
+    public String cancellationTool(Cancellation cancellation) throws InterruptedException {
+        int counter = 0;
+        while (counter++ < 5) {
+            if (cancellation.check().isRequested()) {
+                throw new OperationCancellationException();
+            }
+            TimeUnit.MILLISECONDS.sleep(500);
```

Comment:
Probably makes sense to do the sleep before the check
```suggestion
            TimeUnit.MILLISECONDS.sleep(500);
            if (cancellation.check().isRequested()) {
                throw new OperationCancellationException();
            }
```

Reply:
Done